java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -72,6 +72,15 @@ public UrlDevice getUrlDeviceById(String id) {
   }
 
   /**
+   * Fetches cached URL metadata using the URL broadcasted by the Physical Web device.
+   * @param broadcastUrl The URL broadcasted by the device.
+   * @return Cached metadata relevant to the given URL.
+   */
+  public PwsResult getMetadataByBroadcastUrl(String broadcastUrl) {
+    return mBroadcastUrlToPwsResultMap.get(broadcastUrl);
```

Comment:
Yea, I think a future would be good here.  That will make a good follow-up change.